### DISCUSSION TITLE
MikroORM with MongoDB ?

### DISCUSSION THREAD

**Discussion by souravdasslg** [mikro-orm/mikro-orm#6104] [Q&A] [answered]
I'm considering using MikroORM with MongoDB for a large OTT project. The project is based on NestJS, and we're currently using `@nestjs/mongoose` to define schemas.

Before we make the switch to MikroORM, I'd like to understand a few things:

1. How does MikroORM perform with MongoDB, especially for complex schemas with nested structures and relationships?
2. Are there any limitations or issues to be aware of when using MikroORM with MongoDB compared to SQL databases?
3. For those with experience with Mongoose and MikroORM with MongoDB, how do they compare in terms of ease of use, performance, and features?
4. How comprehensive is MikroORM's documentation when it comes to MongoDB-specific use cases?

I would appreciate any insights, experiences, or resources you can share.

**Comment by B4nan** [comment, marked as answer]
> How does MikroORM perform with MongoDB, especially for complex schemas with nested structures and relationships?

Hard to guess, I am not aware of anybody complaining.

> Are there any limitations or issues to be aware of when using MikroORM with MongoDB compared to SQL databases?

Yes, no joins, so you can't filter by relations. Workaround is to query from the other end, first load the child based on conditions, then load its parent.

> For those with experience with Mongoose and MikroORM with MongoDB, how do they compare in terms of ease of use, performance, and features?

Never used mongoose, can't comment on that.

> How comprehensive is MikroORM's documentation when it comes to MongoDB-specific use cases?

Not much. But you can always fallback to working with the native mongo collection and use `em.map()` to translate the raw results to entities.

**Comment by 2coo** [comment]
I am also planning to migrate from "prisma" to another ORM then this comes to MikroORM. 
Unfortunately, MikroORM does not support time-series collection at this moment.
It seems everything is in it but time-series.
